src/client/Client.cc

```diff
@@ -5413,7 +5385,7 @@ int Client::authenticate()
 {
   assert(client_lock.is_locked_by_me());
 
-  if (authenticated) {
+  if (monclient->is_authenticated()) {
```

Comment:
I'm not sure it is a no-op: the check at the start of ::authenticate is for active_con, rather than authenticated, so if there's no open conenction at that time, it's going to go and wait on the auth_cond even if it's already authenticated.